### ISSUE TITLE
Cannot run `npm version` if package-lock is .gitignore-d

### ISSUE THREAD

**Issue by lennym** [npm/npm#18969]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

Running `npm version` with a local `package-lock.json` file - but with that file in `.gitignore` causes an error when [npm attempts to `git add` the file](https://github.com/npm/npm/blob/latest/lib/version.js#L312).

<details>
```
npm ERR! code 1
npm ERR! Command failed: git add /Users/lenny/path/to/my/code/package-lock.json
npm ERR! The following paths are ignored by one of your .gitignore files:
npm ERR! package-lock.json
npm ERR! Use -f if you really want to add them.
```
</details>

I'm not certain what the logic was at https://github.com/npm/npm/commit/542f7561d173eca40eb8d838a16a0ed582fef989 that meant that the `git add` went from ignoring failures to throwing was, but it introduced this issue. Are there other errors that might be thrown by `git add` that _should_ cause the `version` command to fail? 

I'm aware that "best practice" is considered to be to publish package-lock.json files to version control, but in certain cases I don't feel this is appropriate, and certainly don't think it should cause other commands to crash out with an error.

#### How can the CLI team reproduce the problem?

Run `npm version` in any repo with a package-lock.json file, and with package-lock.json in .gitignore.

### supporting information:

 - `npm -v` prints: v5.5.1 (issue was introduced at https://github.com/npm/npm/commit/542f7561d173eca40eb8d838a16a0ed582fef989 so exists in all versions since 5.3.0)
 - `node -v` prints: v8.1.3

**Comment by lennym** [comment]
Looking further back into the reasoning for https://github.com/npm/npm/commit/53a40cdc1713ef824e03d2e76094e1abb22d3bbc it seems that the fix for [the issue at hand](https://github.com/npm/npm/issues/17707#issuecomment-314485931) was going from parallel to serial execution, and that failing on error was effectively debug information.

It looks like there's basically two options:

* revert the `ignoreError` parameter to `true` for package-lock
* add a check to see if the files are .gitignore-d before trying to commit them

**Comment by ikokostya** [comment]
I have the same problem. Looks like `npm version` command commits `package-lock.json` only if exists.
I fix this by adding `package-lock=false` in `.npmrc` and removing existing `package-lock.json`:

```sh
$ echo "package-lock=false" >> .npmrc # This command helps to not create package-lock.json during npm install
$ rm package-lock.json
$ npm version patch
```

**Comment by corvinrok** [comment]
I agree, this is not a proper way to fall through on this failure. My thoughts on what to do is as follows:

If the ```package-lock.json``` is in the ```.gitignore``` then the ```npm version``` command should try to ignore the ```package-lock.json``` when it tries to execute the ```git add .``` . This all just makes rational sense.

**Comment by RyanZim** [comment]
bump
